spec/functions_spec.rb

```diff
@@ -0,0 +1,468 @@
+require 'spec_helper'
+
+describe SymEngine do
+  before :each do
+  end
+
+  describe 'Function' do
+    before :each do
+      @pi = SymEngine::PI
+      @e = SymEngine::E
+      @one = SymEngine::Integer.new(1)
+      @zero = SymEngine::Integer.new(0)
+      @two = SymEngine::Integer.new(2)
+      @minone = SymEngine::Integer.new(-1)
```

Comment:
Do you need these? You have already used `sympify` inside and also in the tests you have directly used the ruby types like `@pi/2` etc. Remove the variables for 0, 1, 2 and -1 and use Ruby Integers because that is the way it should work and the more the errors appear in tests the better.

You have added tests that give only symengine types as parameters. You shouldn't have to explicitly do these for types like integer that have a direct mapping to types in Ruby. Please keep a test like `sin(1)` where coercion doesn't happen before it is passed.